site/app/controllers/calendar/CalendarController.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace app\controllers\calendar;
+
+use app\controllers\AbstractController;
+use app\controllers\GlobalController;
+use app\libraries\routers\AccessControl;
+use app\libraries\response\JsonResponse;
+use app\libraries\response\WebResponse;
+use app\libraries\response\MultiResponse;
+use app\models\Course;
+use app\models\gradeable\Gradeable;
+use app\models\gradeable\GradeableList;
+use app\models\User;
+use app\views\calendar\CalendarView;
+use Exception;
+use Symfony\Component\Routing\Annotation\Route;
+
+class CalendarController extends AbstractController {
+    private $gradeables = [];
```

Comment:
I think storing a list of gradeables and populating it from the db is something a model should do.
I would suggest the controller on `viewCalendar` will construct a new calendar object which will call `loadFromDb` from its constructor. Then you can just do $calendar->gradeable_list in `viewCalendar` when returning stuff to the view.